tests/functional/test_snap_dcgm.py

```diff
@@ -1,86 +1,175 @@
+
+        :param metric_file: The metric file to check for, if empty check if nothing is loaded
+        """
+        result = subprocess.check_output(
+            "ps -eo cmd | grep '/bin/[d]cgm-exporter'", shell=True, text=True
+        )
+
+        if metric_file:
+            assert f"-f {metric_file}" in result, f"Metric file {metric_file} is not loaded"
+        else:
+            assert "-f" not in result.split(), "Metric file is loaded, but should not be"
+
+    @pytest.mark.parametrize(
+        "service, config, new_value",
+        [
+            ("dcgm.dcgm-exporter", "dcgm-exporter-address", ":9466"),
+            ("dcgm.nv-hostengine", "nv-hostengine-port", "5566"),
+        ],
+    )
+    def test_dcgm_bind_config(self, service: str, config: str, new_value: str) -> None:
+        """Test snap bind configuration."""
+        old_value = self.get_config(config)
+
+        # Valid config
+        self.set_config(service, config, new_value)
+        self.check_bind_config(service, new_value)
+
+        # Invalid config
+        self.set_config(service, config, "test")
+        _check_service_failed(f"snap.{service}")
+
+        # Revert back
+        self.set_config(service, config, old_value)
+        self.check_bind_config(service, old_value)
+
+    def test_dcgm_metric_config(self) -> None:
+        """Test the metric file configuration of the dcgm-exporter service."""
+        service = "dcgm.dcgm-exporter"
+        config = "dcgm-exporter-metrics-file"
+        metric_file = "test-metrics.csv"
+        endpoint = "http://localhost:9400/metrics"
+        metric_file_path = os.path.join("/var/snap/dcgm/common", metric_file)
+
+        self.get_config(config)
+
+        # $SNAP_COMMON requires root permissions to create a file
+        # Will be removed during the test cleanup by the fixture
+        subprocess.check_call(f"sudo touch {metric_file_path}".split())
+
+        # Empty metric
+        self.set_config(service, config, metric_file)
+        self.check_metric_config()
+        _check_endpoint(endpoint)
+
+        # Non-existing metric
+        self.set_config(service, config, "unknown.csv")
+        self.check_metric_config()
+        _check_endpoint(endpoint)
+
+        # Invalid metric
+        subprocess.check_call(f"echo 'test' | sudo tee {metric_file_path}", shell=True)
+        self.set_config(service, config, metric_file)
+        # The exporter will fail to start due to the invalid metric file
+        _check_service_failed(f"snap.{service}")
+
+        # Valid metric
+        subprocess.check_call(
+            f"echo 'DCGM_FI_DRIVER_VERSION, label, Driver Version' | sudo tee {metric_file_path}",
+            shell=True,
+        )
+        self.set_config(service, config, metric_file)
+        self.check_metric_config(metric_file_path)
+        _check_endpoint(endpoint)
+
+        # Revert back
+        self.set_config(service, config)
+        self.check_metric_config()
```

Comment:
these should all be separate test cases. Very generally speaking, you want one test case per assertion, with fresh fixtures for each of them to eliminate risk of test dependencies. In this case I wouldn't go as far as wiping and reinstalling the snap every time, but having separate test cases would help pinpoint to the exact issue when one of them will fail in the future.

Also please add more comments for the empty and non-existent metrics test; it's not obvious why you expect `_check_endpoint` to succeed.